python/modules/IcePy/Types.cpp

```diff
@@ -2504,13 +2500,11 @@ IcePy::SequenceInfo::unmarshalPrimitiveSequence(const PrimitiveInfoPtr& pi, Ice:
         IceUtil::ScopedArray<Ice::Long> arr;
         is->read(p, arr);
         int sz = static_cast<int>(p.second - p.first);
-        if(sm->factory.get())
+        if(sm->factory)
         {
-            result = createSequenceFromMemory(sm,
-                                              reinterpret_cast<const char*>(arr.get() != 0 ? arr.release() : p.first),
-                                              sz * 8,
-                                              BuiltinTypeLong,
-                                              arr.get() != 0);
+            bool adopt = arr.get() != 0;
+            const char* data = reinterpret_cast<const char*>(arr.get() != 0 ? arr.release() : p.first);
+            result = createSequenceFromMemory(sm, data = 0 ? &emptySeq[0] : data, sz * 8, BuiltinTypeLong, adopt);
```

Comment:
`data = 0` isn't correct here. Rather than doing the check in the caller, moving the check for 0 in `createSequenceFromMemory` implementation would be a little less error prone.